### PR TITLE
ID-1131 Refactor GraphQLDispatcher

### PR DESCRIPTION
**Pull request by mickeylee25**
Merge GraphNetwork and GraphQLDispatcher to `GraphQLManager`.

GraphQLManager fetches queries, perform mutations and clear cache via `client: ApolloClient`. The client is initialised in GraphQLFactory and the factory instantiates the GraphQLManager class from CoreFactory.

[BONUS]
- Fixed flaky test on DiskStorageTests
- Resolved TODO on AnalyticsLoggerTests by adding failure test cases

### CODE REVIEW

File: Core/GraphQL/GraphQLManager.swift [hunk 1]
```diff
@@ -48,7 +52,7 @@ public class GraphQLDispatcher: GraphQLDispatching {
         mutation: Query,
```
Comment (You-Hsuan) [inline Core/GraphQL/GraphQLManager.swift:52]: How about we also change the generic parameter naming from `Query` to `Mutation` please? Thank you.

File: Core/CoreFactory.swift [hunk 2]
```diff
@@ -42,19 +42,17 @@ public enum CoreFactory {
...
     ///   - deviceId: Current device UDID set up by UIDeviceInterface protocol
     ///   - reachabilityHelper: ReachabilityChecking generated by makeReachablitiyChecker
-    /// - Returns: GraphQLDispatching protocol
-    public static func makeGraphQLDispatcher(
+    /// - Returns: GraphQLManageable protocol
+    public static func makeGraphQLManager(
         configuration: CoreConfiguration,
         tokenHelper: APITokenManagable,
         deviceId: String,
         reachabilityHelper: ReachabilityChecking
```
Comment (thisolivier) [inline Core/CoreFactory.swift:50]: I notice this reachability helper isn't used any more, and never actually was. Please remove the argument here. 

- [ ] Remove reachabilityHelper

**Review by thisolivier — CHANGES_REQUESTED** [review]

File: Analytics/AnalyticsLogging/AnalyticsLogger.swift [hunk 3]
```diff
@@ -79,6 +79,27 @@ class AnalyticsLogger {
             }
         }
     }
+
+    private func performPutAnalyticEventMutation(
```
[the remark this answers is not in the document]
Comment (mickeylee25) [reply Analytics/AnalyticsLogging/AnalyticsLogger.swift:83]: Yes, we do need the repository/worker layer between GraphQLManager and interface implementation. I've created a [ticket](https://livestyled.atlassian.net/browse/ID-1134) for implementing this on Analytics, Audience and Content, so won't refactor them in this PR.

File: Analytics/Tests/AnalyticsLoggingTests/AnalyticsLoggerTests.swift [hunk 4]
```diff
@@ -56,80 +68,106 @@ final class AnalyticsLoggerTests: XCTestCase {
...
+
+    func test_logEvent_onNotSuccess_delaySingleItem() {
+        let expectation = XCTestExpectation(description: "Event sending completed")
+        mockReachabilityChecker.hasNetworkConnection = true
+        mockGraphQLManager.successReturns = false
+        let sut = makeSut()
+        sut.logEvent(testEvent) { result in
+            switch result {
+            case .success:
+                expectation.fulfill()
```
[the remark this answers is not in the document]
Comment (mickeylee25) [reply Analytics/Tests/AnalyticsLoggingTests/AnalyticsLoggerTests.swift:106]: The test case does go to the success case but I should check if the boolean value is successful or not. I'll update the PR

**Review by You-Hsuan — APPROVED** [review]

**Review by thisolivier — APPROVED** [review]

File: Analytics/Tests/AnalyticsLoggingTests/AnalyticsLoggerTests.swift [hunk 5]
```diff
@@ -39,99 +54,146 @@ final class AnalyticsLoggerTests: XCTestCase {
...
+                XCTAssertTrue(isSuccess)
+                expectation.fulfill()
+            case .failure:
+                XCTFail("We should have sucessfully sent this")
+            }
+        }
+        wait(for: [expectation], timeout: 0.01)
+    }
+
+    func test_logEvent_onNotSuccess_delaySingleItem() {
```
Comment (SoaringEarth) [inline Analytics/Tests/AnalyticsLoggingTests/AnalyticsLoggerTests.swift:118]: could call this ...onFailure_delaySingleItem()

**Pull request merged**